modules/imgproc/src/resize.cpp

```diff
@@ -50,12 +50,899 @@
 #include "precomp.hpp"
 #include "opencl_kernels_imgproc.hpp"
 #include "hal_replacement.hpp"
+#include "opencv2/core/hal/intrin.hpp"
 
 #include "opencv2/core/openvx/ovx_defs.hpp"
 #include "resize.hpp"
 
+#include "opencv2/core/softfloat.hpp"
+
 using namespace cv;
 
+#ifndef CV_ALWAYS_INLINE
+    #if defined(__GNUC__) && (__GNUC__ > 3 ||(__GNUC__ == 3 && __GNUC_MINOR__ >= 1))
+        #define CV_ALWAYS_INLINE inline __attribute__((always_inline))
+    #elif defined(_MSC_VER)
+        #define CV_ALWAYS_INLINE __forceinline
+    #else
+        #define CV_ALWAYS_INLINE inline
+    #endif
+#endif
+
+namespace
+{
+
+class fixedpoint64
+{
```

Comment:
I believe these 250 lines with fixedpoint implementation can be moved into separate .inl.hpp file (in `src` directory)